pennylane/_qubit_device.py

```diff
@@ -759,9 +759,15 @@ def expval(self, observable, shot_range=None, bin_size=None):
             )
             return probs[idx]
 
+        # exact expectation value
         if self.shots is None:
-            # exact expectation value
-            eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            try:
+                eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            except NotImplementedError as e:
+                raise ValueError(
+                    f"Cannot compute analytic expectations of {observable.name} observables."
```

Comment:
```suggestion
                    f"Cannot compute analytic expectations of {observable.name}."
```
Didn't notice before, just for consistency with the other error messages.